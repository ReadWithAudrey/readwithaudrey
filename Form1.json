{"type":"PR","pr":{"number":76,"head":"cc26da92cf5b79d6267b92f31d3ae50d067d061c"},"user":"readwithaudrey.data","status":"pending_review","branch":"cms/Form1","collection":"forms","title":"No Title","description":"No Description!","objects":{"entry":{"path":"src/markdown/formPart1.md","sha":"4fb52fab38e1c19e0df590bd000a4ab8da4290ac"},"files":[]},"timeStamp":"2018-10-10T12:08:41.299Z"}