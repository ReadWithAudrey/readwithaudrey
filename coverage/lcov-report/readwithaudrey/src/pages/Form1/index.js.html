<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/pages/Form1/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/pages/Form1</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Link, graphql } from 'gatsby'
import PropTypes from 'prop-types'
&nbsp;
import {
  InputBox,
  Label,
  Button,
  Layout,
  StatusBar,
  TextBox,
} from '../../components/'
&nbsp;
const Form1 = ({ data }) =&gt; {
  const { q1, q2, q3 } = data.markdownRemark.frontmatter
  return (
    &lt;Layout&gt;
      &lt;form method="POST" action="http://localhost:5000/formPart1"&gt;
        &lt;h1 className="f2 pink tc montserrat mb3 mt4"&gt;Your Details&lt;/h1&gt;
        &lt;StatusBar type="active"&gt;1. Basic Details&lt;/StatusBar&gt;
        &lt;Link to="/Form2"&gt;
          &lt;StatusBar&gt;2. Further Details&lt;/StatusBar&gt;
        &lt;/Link&gt;
        &lt;Link to="/Form3"&gt;
          &lt;StatusBar&gt;3. Your Bio&lt;/StatusBar&gt;
        &lt;/Link&gt;
        &lt;TextBox&gt;
          Welcome to Audrey, thank you for taking the steps to join our
          community.
        &lt;/TextBox&gt;
        &lt;Label&gt;{q1}&lt;/Label&gt;
        &lt;InputBox placeholder="Audrey" /&gt;
        &lt;Label&gt;{q2}&lt;/Label&gt;
        &lt;InputBox placeholder="Readerson" /&gt;
        &lt;Label&gt;{q3}&lt;/Label&gt;
        &lt;InputBox placeholder="audrey@readwithaudrey.com" /&gt;
        &lt;Link to="/Form2" className="no-underline"&gt;
          &lt;Button type="register"&gt;Continue&lt;/Button&gt;
        &lt;/Link&gt;
      &lt;/form&gt;
      &lt;TextBox&gt;
        &lt;Link to="/" className="no-underline"&gt;
          Go back to the homepage
        &lt;/Link&gt;
      &lt;/TextBox&gt;
    &lt;/Layout&gt;
  )
}
&nbsp;
Form1.propTypes = {
  data: PropTypes.object,
}
&nbsp;
export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 1" } }) {
      frontmatter {
        q1
        q2
        q3
      }
    }
  }
`
&nbsp;
export default Form1
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
