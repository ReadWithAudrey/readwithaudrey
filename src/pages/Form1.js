import React from 'react'
import { Link, graphql } from 'gatsby'

import Layout from '../components/layout'

const Form1 = ({ data }) => {
  const { q1, q2, q3 } = data.markdownRemark.frontmatter
  return (
    <Layout>
      <h1>This is p1 of the form {data.site.siteMetadata.title}</h1>
      <p>Q1: {q1}</p>
      <p>Q2: {q2}</p>
      <p>Q3: {q3}</p>
      <Link to="/">Go back to the homepage</Link>
    </Layout>
  )
}

export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 1" } }) {
      frontmatter {
        q1
        q2
        q3
      }
    }
  }
`

export default Form1
