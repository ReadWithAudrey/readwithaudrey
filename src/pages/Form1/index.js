/* eslint-disable */

import axios from 'axios'
import React from 'react'
import { graphql, navigate } from 'gatsby'
import PropTypes from 'prop-types'
import { withSignupContext } from '../../contexts/contextWrapper'

import {
  InputBox,
  Label,
  Button,
  Layout,
  StatusBar,
  TextBox,
  FormSection,
  ErrorSpan,
} from '../../components/'

class Form1 extends React.Component {
  handleSubmit = event => {
    const { firstName, secondName, emailAddress } = this.props.value
    event.preventDefault()
    axios
      .post(`http://localhost:5000/formPart1`, {
        firstName,
        secondName,
        emailAddress,
      })
      .then(res => {
        console.log(res)
        navigate('/form2/')
      })
  }
  render() {
    const {
      firstName,
      secondName,
      emailAddress,
      updateForm,
      nameErrorSpan,
      nameError,
      surnameError,
      surnameErrorSpan,
      emailError,
      emailErrorSpan,
    } = this.props.value
    const {
      heading,
      description,
      q1,
      q2,
      q3,
      q1Placeholder,
      q2Placeholder,
      q3Placeholder,
      finalText,
    } = this.props.data.markdownRemark.frontmatter
    return (
      <Layout>
        <h1 className="f2 pink tc montserrat mb3 mt4">{heading}</h1>
        <FormSection>
          <StatusBar type="active">1. Contact Details</StatusBar>
          <StatusBar>2. Further Details</StatusBar>
          <StatusBar>3. Your Story</StatusBar>
          <TextBox>{description}</TextBox>
          <form onSubmit={this.handleSubmit}>
            <Label>{q1}</Label>
            <ErrorSpan type={nameErrorSpan}>{nameError}</ErrorSpan>
            <InputBox
              placeholder={q1Placeholder}
              onChange={updateForm}
              name="firstName"
              value={firstName}
            />
            <Label>{q2}</Label>
            <ErrorSpan type={surnameErrorSpan}>{surnameError}</ErrorSpan>
            <InputBox
              placeholder={q2Placeholder}
              onChange={updateForm}
              name="secondName"
              value={secondName}
            />
            <Label>{q3}</Label>
            <ErrorSpan type={emailErrorSpan}>{emailError}</ErrorSpan>
            <InputBox
              placeholder={q3Placeholder}
              onChange={updateForm}
              name="emailAddress"
              value={emailAddress}
            />
            <TextBox>{finalText}</TextBox>
            <Button style="register">Continue</Button>
          </form>
        </FormSection>
      </Layout>
    )
  }
}

Form1.propTypes = {
  data: PropTypes.object,
  value: PropTypes.object,
}

export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 1" } }) {
      frontmatter {
        heading
        description
        finalText
        q1
        q2
        q3
        q1Placeholder
        q2Placeholder
        q3Placeholder
      }
    }
  }
`

export default withSignupContext(Form1)
