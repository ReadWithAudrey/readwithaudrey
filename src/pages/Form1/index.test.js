import React from 'react'
import renderer from 'react-test-renderer'
import Form1 from './index.js'

describe('Form1 page snapshot test', () =>
  it('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'Form Part 1',
        },
      },
      markdownRemark: {
        frontmatter: {
          q1: 'First Name',
          q2: 'Surname',
          q3: 'Email Address',
        },
      },
    }
    const tree = renderer.create(<Form1 data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
