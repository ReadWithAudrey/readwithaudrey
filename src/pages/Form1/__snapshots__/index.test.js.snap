// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form1 page snapshot test renders correctly 1`] = `
Array [
  <header
    className="sc-bdVaJa flex items-center justify-between ph3 montserrat white bg-green w-100 f3 chiVga"
  >
    <div
      className="flex items-center"
      id="header-left"
    >
      <img
        alt="book icon"
        className="h2 ph2"
        src="test-file-stub"
      />
      <Link
        to="/"
      >
        Audrey
      </Link>
    </div>
    <Link
      className="w-third"
      to="/Form1/"
    >
      <button
        className="sc-bZQynM bg-green hover-bg-pink pa2 mv2 white br2 w-100 f5 montserrat krNSkU"
      >
        Join Us
      </button>
    </Link>
  </header>,
  <div
    className="flex flex-column items-center justify-center page-padding ph4"
  >
    <h1
      className="f2 pink tc montserrat mb3 mt4"
    >
      Your Details
    </h1>
    <section
      className="sc-chPdSV b--green jHBcBB"
    >
      <section
        className="sc-dnqmqq fl w-third bg-green montserrat icLxSS sc-htoDjs flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 fnsbPi"
      >
        1. Basic Details
      </section>
      <Link
        to="/Form2"
      >
        <section
          className="sc-htoDjs flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 fnsbPi"
        >
          2. Further Details
        </section>
      </Link>
      <Link
        to="/Form3"
      >
        <section
          className="sc-htoDjs flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 fnsbPi"
        >
          3. Your Bio
        </section>
      </Link>
      <p
        className="sc-jTzLTM montserrat f5 tc lh-copy dDfTnQ"
      >
        Welcome to Audrey, thank you for taking the steps to join our community.
      </p>
      <form
        action="http://localhost:5000/formPart1"
        method="POST"
      >
        <label
          className="sc-ifAKCX pink montserrat fw9 f5 pv2 kXFzZl"
        >
          First Name
        </label>
        <input
          className="sc-bxivhb montserrat mv2 pa2 w-100 hCZtbo"
          placeholder="Audrey"
        />
        <label
          className="sc-ifAKCX pink montserrat fw9 f5 pv2 kXFzZl"
        >
          Surname
        </label>
        <input
          className="sc-bxivhb montserrat mv2 pa2 w-100 hCZtbo"
          placeholder="Readerson"
        />
        <label
          className="sc-ifAKCX pink montserrat fw9 f5 pv2 kXFzZl"
        >
          Email Address
        </label>
        <input
          className="sc-bxivhb montserrat mv2 pa2 w-100 hCZtbo"
          placeholder="audrey@readwithaudrey.com"
        />
        <Link
          className="no-underline"
          to="/Form2"
        >
          <button
            className="sc-gzVnrw bg-pink hover-bg-orange white pa2 br2 w-50 h3 f3 mt4 center giBLup"
          >
            Continue
          </button>
        </Link>
      </form>
    </section>
  </div>,
  <footer
    className="sc-bwzfXH flex flex-column justify-center items-center montserrat bg-green w-100 bt bw2 b--green uTxCW"
  >
    <span
      className="sc-htpNat flex justify-around items-center bg-white black w-100  pa2 jbutSu"
    >
      Â© Audrey
      <Link
        className="no-underline black ph1"
        to="/terms/"
      >
        Terms
      </Link>
      <Link
        className="no-underline black ph1"
        to="/privacy/"
      >
        Privacy
      </Link>
      <Link
        className="no-underline black ph1"
        to="/contact/"
      >
        Contact
      </Link>
    </span>
  </footer>,
]
`;
